src/users/repository/InMemory/index.ts

```diff
@@ -1,24 +1,45 @@
+   import { date, z } from "zod";
 import { User } from "~/types";
 
-const UsersInMemory = [
+const UsersInMemory: Array<User> = [
   {
     age: 20,
-    id: "1",
+    id: "123",
     name: "Nombre",
+    email: "jhonDutton@gmail.com",
     password: "123",
+
+  },{
+    age: 22,
+    id: "13",
+    name: "Name",
+    email: "jhonDutton@gmail.com",
+    password: "123",
+  
+  },{
+    age: 52,
+    id: "117",
+    name: "Master Chief",
+    email: "MasterPechosTitanicos@gmail.com",
+    password: "lovuC0rt4na ",
+  
   },
 ] satisfies User[];
 
-//@TODO: Rename to simple name `create` and `get` and so on, to keep it only as the data access layer
-export const createUser = (user: User) => {
-  UsersInMemory.push(user);
-  return user;
+export const create = (user: User) => {
+  return new Promise<User>((resolve, reject) => {
+    if (UsersInMemory.find((existent) => user.id === existent.id)) {
+      reject("already exist");
+    }
+    UsersInMemory.push(user);
+    resolve(user);
+  });
 };
 
-export const getUser = (id: string) => {
-  return UsersInMemory.find((u) => u.id === id);
+export const get = (id: string) => {
+  return UsersInMemory.find((u: User) => u.id === id);
```

Comment:
Please, don't use abbreviations 
```suggestion
  return UsersInMemory.find((user: User) => user.id === id);
```